src/ops/PromoteOps.ts

```diff
@@ -0,0 +1,1217 @@
+      const outcome = await deletePolicyById(policy._id);
+      logmessages.push(`delete policy ${deleteFilePath}`);
+      logmessages.push(`outcome: ${outcome}`);
+      logmessages.push(' ');
+      verboseMessage(`delete policy ${deleteFilePath}\n`);
+      break;
+    }
+    // These next three object types have deletes written for them, but they are not promotable so we don't worry about effecting them
+    case 'cot': {
+      break;
+    }
+    case 'policyset': {
+      break;
+    }
+    case 'saml': {
+      break;
+    }
+    default: {
+      logmessages.push(
+        `No delete ${deleteFilePath} not setup for type ${type}`
+      );
+      logmessages.push(' ');
+      verboseMessage(
+        `No delete ${deleteFilePath} not setup for type ${type}\n`
+      );
+      break;
+    }
+  }
+  return;
+}
+
+/**
+ * Opens the file and returns the json object two keys down into the config object.
+ * @param filePath
+ */
+function getJsonObjectTwoDown(filePath: string): any {
+  try {
+    const data = fs.readFileSync(filePath, 'utf8');
+    const fileData = JSON.parse(data);
+    const jsonObject = fileData[Object.keys(fileData)[0]];
+    return jsonObject[Object.keys(jsonObject)[0]];
+  } catch {
+    new FrodoError('error in json parsing');
+  }
+}
+
+/**
+ * Opens the file and returns a json object one element down from the top object
+ * @param filePath path to the config file
+ */
+function getJsonObjectOneDown(filePath: string): any {
+  try {
+    const data = fs.readFileSync(filePath, 'utf8');
+    const fileData = JSON.parse(data);
+    return fileData[Object.keys(fileData)[0]];
+  } catch {
+    new FrodoError('error in json parsing');
+  }
+}
+
+/**
+ * Sets the realm for the next command to run on.
+ * @param path sub-path to the config file currently being looked at
+ * @param inRealm if the object is in a realm or not
+ */
+function setRealmFromPath(path: string, inRealm: boolean) {
+  if (inRealm) {
+    let realm = path.substring(
+      path.indexOf('/') + 1,
+      path.indexOf('/', path.indexOf('/') + 1)
+    );
+    realm = getRealmUsingExportFormat(realm);
+    verboseMessage(`realm = ${realm}`);
+    state.setRealm(realm);
+  }
+}
+
+/**
+ * Pulls the path out of the path to the config file we are referencing.
+ * @param path the sub-path to the config file
+ */
+function checkForRealmFromPath(path: string) {
+  const inRealm = path.substring(0, path.indexOf('/')) === 'realm';
+  if (inRealm) {
+    const realm = path.substring(
+      path.indexOf('/') + 1,
+      path.indexOf('/', path.indexOf('/') + 1)
+    );
+    realms.add(realm);
+  } else {
+    realms.add('global');
+  }
+}
+
+/**
+ * These three object types are not promotable so we don't care to import or delete them with changes.
+ * @param path path to the config file we are looking at
+ */
+function checkTypeIsPromotable(path: string): boolean {
+  const type = getTypeFromPath(path);
+  let promotable: boolean;
+  switch (type) {
+    case 'cot': {
```

Comment:
These three cases can be combined into a single case. And honestly, since the switch statement isn't all that big, you could probably just do an if statement if you want to.